pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -39,7 +45,49 @@ type openShiftClusterDynamicValidator struct {
 
 	oc              *api.OpenShiftCluster
 	subscriptionDoc *api.SubscriptionDocument
-	fpAuthorizer    refreshable.Authorizer
+	fpAuthorizer    autorest.Authorizer
+}
+
+func ensureAccessTokenClaims(ctx context.Context, tokenCredential *azidentity.ClientSecretCredential, scopes []string) error {
+	var err error
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+
+	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
```

Comment:
I don't think err needs to be set here, the condition can return errors directly inside the function (like it's already doing) that error should be the error result from wait.PollImmediateUtil instead.

e.g.
```
- var err error
+ err := wait.PollImmediateUntil(...
return err
```

Reply:
The original function I copied this from had a [note warning not to do that](https://github.com/Azure/ARO-RP/blob/f8cfc0338a481cc7ce6f007f18c4348d5217bea2/pkg/util/aad/aad.go#L83-L86).  I don't know who's right, but I'll reproduce that note here and leave the logic in its original form.  I don't want to introduce a potential regression.